### PR TITLE
feat : #RDV-30 take appointment modal link back to front + refacto

### PR DESCRIPTION
**Pull request by niko-amosse**
## Describe your changes

- make refacto
- link back to front
- make design for desktop and mobile version of take appointment modal

## Checklist tests

- retrieve the grids i can access (only)
- can change grid by dropdown -> it change grid infos and available timeslots
- can change week -> swow available timeslots or button to move to next avaible timeslot or empty state

#### Desktop Version

image
image
image
image

#### Mobile Version

image
image
image

## Issue ticket number and link

[RDV-30](https://jira.support-ent.fr/browse/RDV-30)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)
- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/services/api/EmptySplitService/index.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+const delayBaseQuery = async (args: any, api: any, extraOptions: any) => {
+  await new Promise((resolve) => setTimeout(resolve, 0));
```
Comment (niko-amosse) [inline frontend/src/services/api/EmptySplitService/index.ts:4]: sert à  simuler un fetching plus long si on le souhaite pour les tests visuels des loaders

File: frontend/src/components/DaySlots/index.tsx [hunk 2]
```diff
@@ -1,33 +1,49 @@
...
+        ) : (
+          <Box sx={noSlotsStyle}>
+            <CloseIcon />
+          </Box>
+        )
       ) : (
-        <Box sx={noSlotsStyle}>
-          <CloseIcon />
-        </Box>
+        Array.from({ length: nbOfSkeletons }).map((_, index) => (
```
Comment (ThomasSevagen) [inline frontend/src/components/DaySlots/index.tsx:44]: sortir cette construction d'array du return

File: frontend/src/components/UserCard/style.ts [hunk 3]
```diff
@@ -16,7 +15,7 @@ import {
 import { BOLD_FONT, ITALIC_FONT } from "~/styles/fontStyle.constants";
 
 export const WrapperUserCard = styled(Box)<WrapperUserCardProps>(
-  ({ status }) => ({
+  ({ isAvailable }) => ({
```
Comment (ThomasSevagen) [inline frontend/src/components/UserCard/style.ts:18]: pas d'erreur console client sans shouldforward prop ?

File: frontend/src/components/WeekSlots/index.tsx [hunk 4]
```diff
@@ -33,41 +33,44 @@ export const WeekSlots: FC = () => {
 
   const dayErrors = useMemo(() => {
     const errors: Record<DAY, boolean> = {} as Record<DAY, boolean>;
-    for (const day of Object.keys(inputs.weekSlots) as DAY[]) {
-      errors[day] = inputs.weekSlots[day].some(
-        (slot) => slots.ids.includes(slot.id) && (!slot.begin || !slot.end),
+    Object.entries(inputs.weekSlots).map(([day, timeSlots]) => {
```
Comment (ThomasSevagen) [inline frontend/src/components/WeekSlots/index.tsx:36]: tu dois avoir un problème dans ta chaîne de typage quele part pour utiliser autant de  `day as DAY`

File: frontend/src/hooks/useBlurGridInputs.ts [hunk 5]
```diff
@@ -49,7 +49,7 @@ export const useBlurGridInputs = (
   const newSlotsError = {
```
Comment (ThomasSevagen) [inline frontend/src/hooks/useBlurGridInputs.ts:49]: il y a moyen de pas faire 4 parcours d'array ici, du moins après le flatmap. JE TE LAISSE DEVINER QUELLE FONCTION JS UTILISER

File: frontend/src/hooks/useUpdateGridInputs/index.ts [hunk 6]
```diff
@@ -138,7 +139,7 @@ export const useUpdateGridInputs: useUpdateGridInputsType = (
 
   const handleAddSlot = (day: DAY) => {
     const newDaySlotsErrorIds = inputs.weekSlots[day]
-      .filter((slots) => !slots.begin || !slots.end)
+      .filter((slots) => !slots.begin.time || !slots.end.time)
```
Comment (ThomasSevagen) [inline frontend/src/hooks/useUpdateGridInputs/index.ts:142]: .filter et .map à la suite on peut transformer ça en un seul parcours. JE TE LAISSE DEVINER QUELLE FONCTION UTILISER

**Review by FlorentMr — APPROVED** [review]

File: frontend/src/providers/TakeAppointmentModalProvider/utils.ts [hunk 7]
```diff
@@ -1,549 +1,73 @@
...
-          end: { hour: 13, minute: 30 },
-        },
-      ],
-    },
-  ],
+export const transformTimeSlotsToDaySlots = (
+  timeSlots: TimeSlots,
+  currentDay: Dayjs,
+): DaySlots[] => {
+  console.log(currentDay.format("YYYY-MM-DD"));
```
Comment (ThomasSevagen) [inline frontend/src/providers/TakeAppointmentModalProvider/utils.ts:32]: log

File: frontend/src/providers/TakeAppointmentModalProvider/utils.ts [hunk 8]
```diff
@@ -1,549 +1,73 @@
...
+  const daySlots: DaySlots[] = Object.values(DAY).map((day) => ({
+    weekDay: day,
+    day: currentDay.startOf("week").add(DAY_VALUES[day].numberOfWeekDay, "day"),
+    slots: [],
+  }));
+
+  console.log(daySlots);
```
Comment (ThomasSevagen) [inline frontend/src/providers/TakeAppointmentModalProvider/utils.ts:39]: log

File: frontend/src/services/api/EmptySplitService/index.ts (hunk 1 above)
Comment (ThomasSevagen) [reply frontend/src/services/api/EmptySplitService/index.ts:4]: ducoup tu vas le laisser?

Comment (niko-amosse) [reply frontend/src/services/api/EmptySplitService/index.ts:4]: je laisse pour l'instant ça me servira pour tester d'autres fetch long sur de nouvelles requetes

**Review by ThomasSevagen — APPROVED** [review]

**Pull request merged**